clients/graphql-kotlin-client/src/main/kotlin/com/expediagroup/graphql/client/GraphQLClientRequest.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.client
+
+/**
+ * Abstract class representing GraphQL request that follows the common GraphQL HTTP request format.
+ *
+ * @see [GraphQL Over HTTP](https://graphql.org/learn/serving-over-http/#post-request) for additional details
+ */
+abstract class GraphQLClientRequest(
```

Comment:
created separate abstract class over using existing `GraphQLRequest` data class as:
* data classes cannot be extended
* need to provide explicit response type to support deserialization of batch responses
* variables are set as `Any?` instead of a `Map` so we can pass arbitrary data classes and they will be automatically serialized as JSON map